java-squid/src/main/java/org/sonar/java/se/NotEqualRelation.java

```diff
@@ -36,25 +43,59 @@ protected String getOperand() {
   }
 
   @Override
-  protected Boolean implies(SymbolicValueRelation relation) {
-    if (hasSameOperand(relation)) {
-      return relation instanceof NotEqualRelation;
-    }
+  protected Boolean isImpliedBy(SymbolicValueRelation relation) {
+    return relation.impliesNotEqual();
+  }
+
+  @Override
+  protected Boolean impliesEqual() {
+    return Boolean.FALSE;
+  }
+
+  @Override
+  protected Boolean impliesNotEqual() {
+    return Boolean.TRUE;
+  }
+
+  @Override
+  @CheckForNull
+  protected Boolean impliesMethodEquals() {
+    return null;
+  }
+
+  @Override
+  @CheckForNull
+  protected Boolean impliesNotMethodEquals() {
     return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Null is returned but a "Boolean" is expected. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS2447)